cardano-api/src/Cardano/Api/Orphans.hs

```diff
@@ -1,12 +1,29 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
I'm wondering if there is a way to avoid using `Undecidableinstances` here? Maybe this could be solved by OVERLAPPING / OVERLAPPABLE?